File: app/src/main/java/com/moldedbits/android/BaseActivity.java
package com.moldedbits.android;

import android.os.Bundle;
import android.os.Message;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;
import android.widget.FrameLayout;

import com.moldedbits.android.api.ApiService;
import com.moldedbits.android.dialogs.LoadingDialog;
import com.moldedbits.android.utils.fragmenttransactionhandler.FragmentTransactionHandler;

import javax.inject.Inject;


public abstract class BaseActivity extends AppCompatActivity {

    @Inject
    ApiService apiService;

    private FrameLayout contentFrame;
    protected FragmentTransactionHandler handler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        super.setContentView(R.layout.activity_base);

        BaseApplication.getInstance().getApiComponent().inject(this);

        contentFrame = (FrameLayout) findViewById(R.id.base_container);
        handler = new FragmentTransactionHandler();
    }

    @Override
    public void setContentView(int layoutResId) {
        getLayoutInflater().inflate(layoutResId, contentFrame, true);
    }

    @Override
    protected void onResume() {
        super.onResume();
        handler.setActivity(this);
        handler.resume();
    }

    @Override
    protected void onPause() {
        super.onPause();
        handler.pause();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        handler.setActivity(null);
    }

    protected FragmentTransactionHandler getHandler() {
        return handler;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    public void dismissLoadingDialogWithHandler(LoadingDialog dialog) {
        Message msg
                = getHandler().obtainMessage(FragmentTransactionHandler.LOADING_DIALOG_DISMISS_MSG,
                dialog);
        getHandler().sendMessage(msg);
    }

}

File: app/src/main/java/com/moldedbits/android/BaseActivity.kt
package com.moldedbits.android

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import android.widget.FrameLayout
import com.moldedbits.android.api.ApiService
import com.moldedbits.android.dialogs.LoadingDialog
import com.moldedbits.android.utils.fragmenttransactionhandler.FragmentTransactionHandler
import javax.inject.Inject


/**
 * Created by shishank
 * on 08/01/16.
 */
abstract class BaseActivity : AppCompatActivity() {

    @Inject
    internal var apiService: ApiService? = null

    private var contentFrame: FrameLayout? = null
    protected lateinit var handler: FragmentTransactionHandler

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        super.setContentView(R.layout.activity_base)

        BaseApplication.instance.apiComponent.inject(this)

        contentFrame = findViewById(R.id.base_container)
        handler = FragmentTransactionHandler()
    }

    override fun setContentView(layoutResId: Int) {
        layoutInflater.inflate(layoutResId, contentFrame, true)
    }

    override fun onResume() {
        super.onResume()
        handler.setActivity(this)
        handler.resume()
    }

    override fun onPause() {
        super.onPause()
        handler.pause()
    }

    override fun onDestroy() {
        super.onDestroy()
        handler.setActivity(null)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                onBackPressed()
                return true
            }
            else -> return super.onOptionsItemSelected(item)
        }
    }

    fun dismissLoadingDialogWithHandler(dialog: LoadingDialog) {
        val msg = handler.obtainMessage(FragmentTransactionHandler.LOADING_DIALOG_DISMISS_MSG,
                dialog)
        handler.sendMessage(msg)
    }

}
